### PR TITLE
Additions to XMarkerListToFile and XMarkerListFromFile

### PR DESCRIPTION
**Pull request by evast**
Extended XMarkerListToFile and XMarkerListFromFile to save all 6 dimensions of the XMarker vector and to specify the decimal precision to use when saving the XMarkers to file.

### CODE REVIEW

File: Community/General/Sources/ML/MLXMarkerListCommunityModules/XMarkerListFile/mlXMarkerListToFile.cpp [hunk 1]
```diff
@@ -225,28 +238,40 @@ void XMarkerListToFile::handleNotification (Field *field)
...
               file_op << coordinateSeparator.str();
             }
           }
+		  if (posS) {
+			file_op << marker.pos[3];
+			if (vecX || vecY || vecZ || type || name) {
+			  file_op << coordinateSeparator.str();
+			}
+		  }
           if (posT) {
```
Comment (hmeine) [inline Community/General/Sources/ML/MLXMarkerListCommunityModules/XMarkerListFile/mlXMarkerListToFile.cpp:263]: Here, the indentation seems to be wrong.

File: Community/General/Sources/ML/MLXMarkerListCommunityModules/XMarkerListFile/mlXMarkerListToFile.cpp [hunk 2]
```diff
@@ -2,8 +2,8 @@
 //! The ML module class XMarkerListToFile.
 /*!
 // \file    mlXMarkerListToFile.cpp
-// \author  Coert Metz
-// \date    2007-07-06
+// \author  Coert Metz, Erwin Vast
+// \date    2007-07-06, 20-11-15
```
Comment (hmeine) [inline Community/General/Sources/ML/MLXMarkerListCommunityModules/XMarkerListFile/mlXMarkerListToFile.cpp]: Don't you prefer the ISO date format (unambiguous for all days/months, alphanumerically sortable, ...)?

File: Community/General/Sources/ML/MLXMarkerListCommunityModules/XMarkerListFile/mlXMarkerListFromFile.cpp [hunk 3]
```diff
@@ -154,15 +158,17 @@ void XMarkerListFromFile::handleNotification (Field *field)
...
+          std::vector<std::string> tokens (10, "0");
           if (_positionXFld->getBoolValue()) if ( !(file_op >> tokens[0]) ) break;
           if (_positionYFld->getBoolValue()) if ( !(file_op >> tokens[1]) ) break;
           if (_positionZFld->getBoolValue()) if ( !(file_op >> tokens[2]) ) break;
-          if (_positionTFld->getBoolValue()) if ( !(file_op >> tokens[3]) ) break;
-          if (_vectorXFld->getBoolValue()) if ( !(file_op >> tokens[4]) ) break;
-          if (_vectorYFld->getBoolValue()) if ( !(file_op >> tokens[5]) ) break;
-          if (_vectorZFld->getBoolValue()) if ( !(file_op >> tokens[6]) ) break;
-          if (_typeFld->getBoolValue()) if ( !(file_op >> tokens[7]) ) break;
+		  if (_positionSFld->getBoolValue()) if ( !(file_op >> tokens[3])) break;
```
Comment (hmeine) [inline Community/General/Sources/ML/MLXMarkerListCommunityModules/XMarkerListFile/mlXMarkerListFromFile.cpp]: indentation?

**Comment by hmeine** [comment]
Reviewed, looks good, except for the indentation.

**Comment by evast** [comment]
Thanks for the review. The indentation has been corrected, also in other parts of the files. Dates now have ISO format. Could you check and pull/merge? Thanks!

**Comment by hmeine** [comment]
Thanks for the contribution, and for fixing the indentation!

**Pull request merged**